### PR TITLE
Update Solution Exporting to the new approach

### PR DESCRIPTION
**Pull request by andreiborodin**
This change goes ahead and updates the remaining export logic into the new approach that seperates responsibilities.

### CODE REVIEW

File: Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectGenerator/Scripts/UnityProjectInfo.cs [hunk 1]
```diff
@@ -478,6 +484,190 @@ private void ScanAndProcessFiles(string folder, Dictionary<string, Action<string
...
 
         #region Export Logic
+        public void ExportSolution(IUnityProjectExporter unityProjectExporter, FileInfo solutionFilePath, DirectoryInfo generatedProjectsFolder)
+        {
+            SolutionFileInfo solutionFileInfo = TextSolutionFileParser.ParseExistingSolutionFile(logger, solutionFilePath.FullName);
+            ISolutionExporter exporter = unityProjectExporter.CreateSolutionExporter(logger, solutionFilePath);
+
+            //TODO we need to figure out how to handle existing projects
+            //Dictionary<string, Guid> existingProjects;
+
```
Comment (chrisfromwork) [inline Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectGenerator/Scripts/UnityProjectInfo.cs]: nit: probably don't need this second line

File: Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectGenerator/Scripts/UnityProjectInfo.cs [hunk 2]
```diff
@@ -478,6 +484,190 @@ private void ScanAndProcessFiles(string folder, Dictionary<string, Action<string
...
+            // Bring forward the properties, then set the one we care about
+            exporter.Properties.AddRange(solutionFileInfo.Properties);
+            exporter.Properties["HideSolutionNode"] = "FALSE";
+
+            // Bring forward the extensibility globals, then set the one we care about
+            exporter.ExtensibilityGlobals.AddRange(solutionFileInfo.ExtensibilityGlobals);
+            exporter.ExtensibilityGlobals["SolutionGuid"] = "{" + config.SolutionGuid.ToString().ToUpper() + "}";
+
+            // Bring forward the pairs, and then set the platforms we know off
```
Comment (chrisfromwork) [inline Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectGenerator/Scripts/UnityProjectInfo.cs]: >off [](start = 75, length = 3)

nit: of

File: Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectGenerator/Scripts/UnityProjectInfo.cs [hunk 3]
```diff
@@ -478,6 +484,190 @@ private void ScanAndProcessFiles(string folder, Dictionary<string, Action<string
...
+                {
+                    exporter.GetOrAddFolder(guidNameTuple.Item1, guidNameTuple.Item2, isGenerated: true)
+                        .Children.Add(getGuidFunc(item));
+                }
+            }
+        }
+
+        private Dictionary<AssetLocation, Tuple<Guid, string>> GetKnownFolderMapping()
+        {
+            return new Dictionary<AssetLocation, Tuple<Guid, string>>()
```
Comment (chrisfromwork) [inline Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectGenerator/Scripts/UnityProjectInfo.cs]: >Dictionary [](start = 23, length = 10)

nit: can this be a constant or static generated?

File: Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectGenerator/Scripts/Exporters/Extensions.cs.cs [hunk 4]
```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
```
Comment (chrisfromwork) [inline Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectGenerator/Scripts/Exporters/Extensions.cs.cs]: nit: it looks like this file has an extra *.cs extension in its name

File: Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectGenerator/Scripts/Exporters/ISolutionExporter.cs [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+        /// </summary>
+        IDictionary<Guid, SolutionFolder> Folders { get; }
+
+        /// <summary>
+        /// Gets a mutable dictionary of configuration/platform sets for the solution.
+        /// </summary>
+        ISet<ConfigurationPlatformPair> ConfigurationPlatforms { get; }
+        
+        /// <summary>
+        /// Gets a mutable set of items that conisdered generated and will be exported as such.
```
Comment (chrisfromwork) [inline Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectGenerator/Scripts/Exporters/ISolutionExporter.cs]: >conisdered [](start = 45, length = 10)

nit: considered

File: Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectGenerator/Scripts/Exporters/ISolutionExporter.cs [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+        /// </summary>
+        ISet<Guid> GeneratedItems { get; }
+
+        /// <summary>
+        /// Gets or set the mutable dictionary of solution properties to export.
```
Comment (chrisfromwork) [inline Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectGenerator/Scripts/Exporters/ISolutionExporter.cs]: >set [](start = 20, length = 3)

nit, these comments may read better as "Gets or sets"

**Comment by chrisfromwork** [comment]
I left some nits but didn't notice anything worth blocking on.

Does this review work well with existing projects/MSBuildForUnity configured projects? Are there any steps that could be added to this review for migrating to this change + can we get a tag for this review stating their are migration instructions if they are indeed needed?

**Comment by andreiborodin** [comment]
Thanks, and yes it does.

---
In reply to: [579939077](https://github.com/microsoft/MSBuildForUnity/pull/116#issuecomment-579939077) [](ancestors = 579939077)

File: Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectGenerator/Scripts/UnityProjectInfo.cs (hunk 3 above)
Comment (andreiborodin) [reply Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectGenerator/Scripts/UnityProjectInfo.cs]: Nope, cause of the config reference.

---
In reply to: [372602902](https://github.com/microsoft/MSBuildForUnity/pull/116#discussion_r372602902) [](ancestors = 372602902)

File: Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectGenerator/Scripts/Exporters/Extensions.cs.cs (hunk 4 above)
Comment (andreiborodin) [reply Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectGenerator/Scripts/Exporters/Extensions.cs.cs]: Are you saying that this Extensions file that is a .cs class has an extra extension which is .cs? :D

---
In reply to: [372603757](https://github.com/microsoft/MSBuildForUnity/pull/116#discussion_r372603757) [](ancestors = 372603757)

**Pull request merged**